syntax = "proto3";

package proto;

option go_package = "github.com/Jorrit05/DYNAMOS/pkg/proto";
import "google/protobuf/empty.proto";
import "google/protobuf/struct.proto";

// The sidecar definition.
service Microservice {
  rpc InitMicroServiceServer(google.protobuf.Empty) returns (google.protobuf.Empty) {}
  rpc SendData(MicroserviceCommunication) returns (google.protobuf.Empty) {}
}


message MicroserviceCommunication {
  string type = 1;
  google.protobuf.Struct data = 2;
  map<string, string> metadata = 3;
}
